cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -238,197 +562,65 @@ instance FromJSON ProtocolParameters where
-      , protocolUpdateTxFeePerByte        = merge protocolUpdateTxFeePerByte
-      , protocolUpdateMinUTxOValue        = merge protocolUpdateMinUTxOValue
-      , protocolUpdateStakeAddressDeposit = merge protocolUpdateStakeAddressDeposit
-      , protocolUpdateStakePoolDeposit    = merge protocolUpdateStakePoolDeposit
-      , protocolUpdateMinPoolCost         = merge protocolUpdateMinPoolCost
-      , protocolUpdatePoolRetireMaxEpoch  = merge protocolUpdatePoolRetireMaxEpoch
-      , protocolUpdateStakePoolTargetNum  = merge protocolUpdateStakePoolTargetNum
-      , protocolUpdatePoolPledgeInfluence = merge protocolUpdatePoolPledgeInfluence
-      , protocolUpdateMonetaryExpansion   = merge protocolUpdateMonetaryExpansion
-      , protocolUpdateTreasuryCut         = merge protocolUpdateTreasuryCut
-      }
-      where
-        -- prefer the right hand side:
-        merge :: (ProtocolParametersUpdate -> Maybe a) -> Maybe a
-        merge f = f ppu2 `mplus` f ppu1
-
-instance Monoid ProtocolParametersUpdate where
-    mempty =
-      ProtocolParametersUpdate {
-        protocolUpdateProtocolVersion     = Nothing
-      , protocolUpdateDecentralization    = Nothing
-      , protocolUpdateExtraPraosEntropy   = Nothing
-      , protocolUpdateMaxBlockHeaderSize  = Nothing
-      , protocolUpdateMaxBlockBodySize    = Nothing
-      , protocolUpdateMaxTxSize           = Nothing
-      , protocolUpdateTxFeeFixed          = Nothing
-      , protocolUpdateTxFeePerByte        = Nothing
-      , protocolUpdateMinUTxOValue        = Nothing
-      , protocolUpdateStakeAddressDeposit = Nothing
-      , protocolUpdateStakePoolDeposit    = Nothing
-      , protocolUpdateMinPoolCost         = Nothing
-      , protocolUpdatePoolRetireMaxEpoch  = Nothing
-      , protocolUpdateStakePoolTargetNum  = Nothing
-      , protocolUpdatePoolPledgeInfluence = Nothing
-      , protocolUpdateMonetaryExpansion   = Nothing
-      , protocolUpdateTreasuryCut         = Nothing
-      }
-
+                        <*> o .:? "utxoCostPerByte"
+                        <*> o .:? "costModel" .!= mempty
+                        <*> o .:? "executionUnitPrices" .!= mempty
+                        <*> o .:? "maxTxExecUnits"
+                        <*> o .:? "maxBlockExecUnits"
+                        <*> o .:? "maxValueSize"
+           case eParams of
+             Left err -> fail . Text.unpack $ renderProtocolParamsErr err
+             Right pparams -> return pparams
+
+
+instance IsCardanoEra era => ToJSON (ProtocolParameters era) where
+  toJSON pp =
+    case cardanoEraStyle cardanoEra :: CardanoEraStyle era of
+      LegacyByronEra -> error "Protocol parameters are not supported in the Byron era."
+      ShelleyBasedEra sbe -> createProtocolParametersObject sbe pp
+
+createProtocolParametersObject
+  :: ShelleyBasedEra era  -> ProtocolParameters era -> Aeson.Value
+createProtocolParametersObject sbe pp =
```

Comment:
So this would not exist at all. We'd just parse them all, like in the original `FromJSON` instance.